components/FilterLeadsCard/FilterLeadsCard.tsx

```diff
@@ -0,0 +1,236 @@
+
+  LEADS_DATA.map((item) =>
+    createdByUsersList.push({
+      value: item.assignedByName,
+      label: item.assignedByName,
+    })
+  );
+
+  LEADS_DATA.map((item) =>
+    assigneeUsersList.push({
+      value: item.assignedToName,
+      label: item.assignedToName,
+    })
+  );
+
+  //* Output
+  const filterData = {
+    startDate: startDate,
+    endDate: endDate,
+    status: status,
+    createdBy: selectedCreatedBy,
+    assignee: selectedAssignee,
+  };
+
+  const handleApplyFilterButton = () => {
+    if (
+      filterData.createdBy === null &&
+      filterData.assignee === null &&
+      filterData.status.hot === false &&
+      filterData.status.warm === false &&
+      filterData.status.cold === false &&
+      filterData.endDate === null
+    ) {
+      setFilterCardOpen(false);
+      return null;
+    } else {
+      onFilterData(filterData);
+      setFilterCardOpen(false);
+    }
+  };
+
+  const handleCancelFilterButton = () => {
+    setFilterCardOpen(false);
+  };
+
+  return (
+    <div className='relative text-left bg-white shadow-md'>
+      {/* Dropdown Content */}
+      <div className='absolute z-10 w-[389px] h-[452px] rounded-xl shadow-lg bg-white ring-1 ring-black ring-opacity-5 right-0'>
+        <div className='p-[20px]'>
+          {/* Created by */}
+          <div className='my-2'>
+            <label className='font-semibold text-[#00156A] text-[16px] mb-1'>
+              Created by
+            </label>
+            <div className=''>
+              <div className=''>
+                <div>
+                  <CustomMultiSelect
+                    setSelected={setCreatedBy}
+                    options={createdByUsersList}
+                    onSelectChange={(value:any) => setCreatedBy(value)}
+                  />
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <div className='border-t my-[10px] w-[349px] h-[1px] border-[#E9F0FF]'></div>
+
+          {/* Assignee */}
+          <div className='my-2'>
+            <label className='font-semibold text-[#00156A] text-[16px] mb-1'>
+              Assignee
+            </label>
+            <div className=''>
+              <div className=' items-center'>
+                <div>
+                  <CustomMultiSelect
+                    setSelected={setAssignee}
+                    options={assigneeUsersList}
+                    onSelectChange={(value:any) => setAssignee(value)}
+                  />
+                </div>
+              </div>
+            </div>
+          </div>
+
+          <div className='border-t my-[10px] w-[349px] h-[1px] border-[#E9F0FF]'></div>
+
+          {/* Status */}
+          <div className='my-2'>
+            <label className='font-semibold text-[#00156A] text-[16px] mb-1'>
+              Status
+            </label>
+            <div className='flex ml-1 my-1'>
+              <div className='flex items-center me-4'>
```

Comment:
I believe we can create a component for this portion as these are similar for Hot, Warm and Cold status. We can give a name for this components as `StatusCheckbox`.